fusedrug_examples/interaction/drug_target/affinity_prediction/t5_mol/tokenizer/multi_tokenizer.py

```diff
@@ -0,0 +1,1304 @@
+
+            all_inds_set = all_inds_set.union(special_tokens_ID_set)
+            if len(all_inds_set) != all_inds_len + len(set(special_tokens_ID_set)):
+                result["ID collisions across vocabs"] = False
+                result_details["ID collisions across vocabs"].append("special")
+            all_inds_len = len(all_inds_set)
+
+        return result, result_details
+
+    def is_consistent(self) -> bool:
+        """Returns True if the modular tokenizer is consistent, i.e.:
+        a. Special tokens are the same (and map to the same indices) across all the tokenizers
+        b. Regular token ID mappings of any given tokenizer do not collide with special token mappings, nor with ID mappings of other tokenizers
+
+        Raises:
+            Exception: _description_
+
+        Returns:
+            bool: _description_
+        """
+        raise Exception("Not implemented")
+
+    @staticmethod
+    def cfg_list_2_dict(dict_list: List) -> Dict[str, Any]:
+        """Receives a list of dicts, each containing a key "name" and changes it to
+
+        Args:
+            dict_list (List): _description_
+
+        Returns:
+            Dict[str, Any]: _description_
+        """
+        return {d["name"]: d for d in dict_list}
+
+    def save_jsons(self, tokenizers_info: Optional[List] = None) -> None:
+        """_summary_
+
+        Args:
+            tokenizers_info_list (Optional[List], optional): A list of dictionaries containing the following:
+            [
+                   {
+                        "name": a name of a tokenizer
+                        "modular_json_path":out_path for tokenizer_type
+                    }
+            ]
+            In case of None, paths stored in self.tokenizers_info (modular_json_path for each tokenizer) are used.
+            In case of partial tokenizer_types, only those tokenizers will be saved
+            Defaults to None.
+            TODO: also save the config yaml there
+        """
+        if tokenizers_info is None:
+            for t_type in self.tokenizers_info:
+                tokenizer_inst = self.tokenizers_info[t_type]["tokenizer_inst"]
+                out_path = self.tokenizers_info[t_type]["modular_json_path"]
+                tokenizer_inst.save(out_path)
+        else:
+            tokenizers_info_dict = ModularTokenizer.cfg_list_2_dict(tokenizers_info)
+            for t_type in tokenizers_info_dict:
+                tokenizer_inst = self.tokenizers_info[t_type]["tokenizer_inst"]
+                out_path = tokenizers_info_dict[t_type]["modular_json_path"]
+                tokenizer_inst.save(out_path)
+
+    def _add_single_tokenizer(
+        self,
+        tokenizer_info: Dict,
+    ) -> None:
+        raise Exception("Not implemented")
+
+    def add_tokenizers(
+        self,
+    ) -> None:
+        raise Exception("Not implemented")
+
+    def _encode_single_type(self, data_str: str, input_type: str, sequence_id: Optional[int] = None) -> Encoding:
```

Comment:
What is sequence_id and why do we need it?

Reply:
sequence_id is used by the original tokenizer to mark different sentences, so it can be useful to mark tokens encoded by different tokenizers.